architecture/protocol.go

```diff
@@ -147,6 +147,16 @@ func (command *Command) Parse(rawCommand string) (bool, error) {
 			return true, command.Err
 		}
 
+		// Check if parameter types are correct
+		for i, _ := range opts.Params {
+			if opts.ParamInteger[i] {
+				_, err := strconv.Atoi(parts[i+1])
```

Comment:
I would like to see these parsed parameters become part of the command object so that the parsing should only be done once